File: examples/image-generation/openai_sdk_example.py
"""
Пример использования генерации изображений через OpenAI Python SDK
Полная совместимость с openai-python библиотекой
"""

from openai import OpenAI
import os
from datetime import datetime

# Настройка клиента для вашего API
client = OpenAI(
    base_url='http://localhost:3000/v1',  # URL вашего API
    api_key='your-api-key'  # Или из переменной окружения OPENAI_API_KEY
)

def generate_image():
    """Генерация изображения через OpenAI SDK"""
    try:
        print('🎨 Генерация изображения через OpenAI Python SDK...')
        
        response = client.images.generate(
            prompt='A beautiful sunset over the ocean with dramatic clouds',
            model='dall-e-3',  # Будет замаплено на qwen-image-max
            size='1024x1024',
            n=1
        )
        
        print('✅ Изображение сгенерировано!')
        print(f'URL: {response.data[0].url}')
        print(f'Revised prompt: {response.data[0].revised_prompt}')
        print(f'Created: {datetime.fromtimestamp(response.created).isoformat()}')
        
        return response.data[0].url
        
    except Exception as error:
        print(f'❌ Ошибка: {error}')
        return None

def list_image_models():
    """Получение списка моделей для генерации изображений"""
    try:
        print('\n📋 Список моделей для генерации изображений:')
        
        models = client.models.list()
        image_models = [
            m for m in models.data 
            if hasattr(m, 'capabilities') and 'image_generation' in m.capabilities
        ]
        
        for model in image_models:
            print(f'  - {model.id} ({model.owned_by})')
            
    except Exception as error:
        print(f'❌ Ошибка: {error}')

def generate_multiple_images():
    """Генерация нескольких изображений"""
    try:
        print('\n🎨 Генерация нескольких изображений...')
        
        response = client.images.generate(
            prompt='A cute cat sitting on a windowsill looking outside',
            model='qwen-image-plus',
            size='1024x1024',
            n=1  # Можно увеличить, если API поддерживает
        )
        
        print(f'✅ Сгенерировано {len(response.data)} изображений:')
        for i, image in enumerate(response.data, 1):
            print(f'  {i}. {image.url}')
        
    except Exception as error:
        print(f'❌ Ошибка: {error}')

if __name__ == '__main__':
    # Запуск примеров
    generate_image()
    list_image_models()
    generate_multiple_images()

File: examples/image-generation/openai-sdk-example.js
// Пример использования генерации изображений через OpenAI SDK
// Полная совместимость с openai-node библиотекой

import OpenAI from 'openai';

// Настройка клиента для вашего API
const openai = new OpenAI({
    baseURL: 'http://localhost:3000/v1', // URL вашего API
    apiKey: 'your-api-key' // Или из .env
});

async function generateImage() {
    try {
        console.log('🎨 Генерация изображения через OpenAI SDK...');

        const image = await openai.images.generate({
            prompt: 'A beautiful sunset over the ocean with dramatic clouds',
            model: 'dall-e-3', // Будет замаплено на qwen-image-max
            size: '1024x1024',
            n: 1
        });

        console.log('✅ Изображение сгенерировано!');
        console.log('URL:', image.data[0].url);
        console.log('Revised prompt:', image.data[0].revised_prompt);
        console.log('Created:', new Date(image.created * 1000).toISOString());

    } catch (error) {
        console.error('❌ Ошибка:', error.message);
    }
}

async function listImageModels() {
    try {
        console.log('\n📋 Список моделей для генерации изображений:');

        const models = await openai.models.list();
        const imageModels = models.data.filter((m) =>
            m.capabilities?.includes('image_generation')
        );

        imageModels.forEach((model) => {
            console.log(`  - ${model.id} (${model.owned_by})`);
        });

    } catch (error) {
        console.error('❌ Ошибка:', error.message);
    }
}

// Запуск
await generateImage();
await listImageModels();
